### PR TITLE
Add solution for #55 Jump Game and #57. Insert Interval

### PR DESCRIPTION
**Pull request by zj9205**

### CODE REVIEW

File: DP/JumpGame.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/**
+ * Question Link: https://leetcode.com/problems/jump-game/
+ * Primary idea: check each position with the previous farest step can reach. If i > last farest step, means cannot reach
+ * Time Complexity: O(n), Space Complexity: O(1)
+ *
+ */
+
+class Solution {
```
Comment (soapyigu) [inline DP/JumpGame.swift]: Please change the class name to JumpGame, thanks

File: DP/JumpGame.swift [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    func canJump(_ nums: [Int]) -> Bool {
+        var max = 0
+        
+        for i in 0 ..< nums.count {
+            var farestStep = i + nums[i]
+            if (i > max) {
```
Comment (soapyigu) [inline DP/JumpGame.swift]: There is no need to add parentheses for if statement

File: DP/JumpGame.swift [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ * Time Complexity: O(n), Space Complexity: O(1)
+ *
+ */
+
+class Solution {
+    func canJump(_ nums: [Int]) -> Bool {
+        var max = 0
+        
+        for i in 0 ..< nums.count {
+            var farestStep = i + nums[i]
```
Comment (soapyigu) [inline DP/JumpGame.swift]: farestStep is not changed in the future, so we should use let instead

File: Sort/InsertInterval.swift [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ * public class Interval {
+ *     public var start: Int
+ *     public var end: Int
+ *     public init(_ start: Int, _ end: Int) {
+ *         self.start = start
+ *         self.end = end
+ *     }
+ * }
+ */
+class Solution {
```
Comment (soapyigu) [inline Sort/InsertInterval.swift]: Could you please change the class name to file name?

File: Sort/InsertInterval.swift [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        var result : [Interval] = []
+        var tempInterval = Interval(newInterval.start, newInterval.end)
+        
+        while index < intervals.count && newInterval.start > intervals[index].end {
+            result.append(intervals[index])
+            index += 1
+        }
+        
+        while index < intervals.count && newInterval.end >= intervals[index].start {
+            var minStart = min(tempInterval.start, intervals[index].start)
```
Comment (soapyigu) [inline Sort/InsertInterval.swift]: minStart and maxEnd are immutable; thus we should use let instead of var

File: Sort/InsertInterval.swift [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+ *     public init(_ start: Int, _ end: Int) {
+ *         self.start = start
+ *         self.end = end
+ *     }
+ * }
+ */
+class Solution {
+    func insert(_ intervals: [Interval], _ newInterval: Interval) -> [Interval] {
+        var index = 0
+        var result : [Interval] = []
```
Comment (soapyigu) [inline Sort/InsertInterval.swift]: var result = [Interval](), please follow Ray Wenderlich's swift code style

File: Sort/InsertInterval.swift [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+ * Definition for an interval.
+ * public class Interval {
+ *     public var start: Int
+ *     public var end: Int
+ *     public init(_ start: Int, _ end: Int) {
+ *         self.start = start
+ *         self.end = end
+ *     }
+ * }
+ */
```
Comment (soapyigu) [inline Sort/InsertInterval.swift]: Please add a blank line between question and solution.

**Comment by zj9205** [comment]
Hi, I finished modifications, please review

**Pull request merged**